observations/obs_converters/AIRS/convert_amsu_L1.f90

```diff
@@ -107,31 +122,81 @@ program convert_amsu_L1
 ! same channels - I think so.
 call channel_list_to_indices(channel_list, use_channels)
 
-! for each input file
+! Figure out how many (more) observations there may be in the output.
+! It is impossible to predict how the lat/lon subsetting may affect this
+! number, but the thinning and channels can be taken into account.
+
+max_num = max_possible_obs(filecount, cross_track_thin, along_track_thin, use_channels)
+
+! either read existing obs_seq or create a new one
+
+inquire(file=outputfile, exist=file_exist)
+
+if ( file_exist .and. append_output ) then
+  call read_obs_seq(outputfile, 0, 0, max_num, big_sequence)
+  write(string1,*)'Appending to "'//trim(outputfile)//'"'
+  write(string2,*)'Initially has ',get_num_obs(big_sequence),' observations.'
+else
+  call init_obs_sequence(big_sequence, NUM_COPIES, NUM_QC, max_num)
+  do i = 1, NUM_COPIES
+    call set_copy_meta_data(big_sequence, i, 'observation')
+  end do
+  do i = 1, NUM_QC
+    call set_qc_meta_data(big_sequence, i, 'QC')
+  end do
+  write(string1,*)'Creating "'//trim(outputfile)//'" from scratch.'
+  write(string2,*)'Initially has ',get_num_obs(big_sequence),' observations.'
+endif
+
+if (verbose > 0) call error_handler(E_MSG,source,string1,text2=string2)
+
+! read from netCDF file into a derived type that holds all the information
+! convert derived type information to DART sequence for that file
+! combine that sequence to the output sequence
+! release the file sequence memory in preparation for creating it new next iteration
+
 do ifile = 1,filecount
 
-   ! read from HDF file into a derived type that holds all the information
    call read_amsua_bt_netCDF_granule(l1_files(ifile), granule)   
 
-   ! convert derived type information to DART sequence
-   call make_obs_sequence(seq, granule, lon1, lon2, lat1, lat2, &
+   call make_obs_sequence(small_sequence, granule, lon1, lon2, lat1, lat2, &
                           use_channels, along_track_thin, cross_track_thin)
 
-   ! write the sequence to a disk file
-   call write_obs_seq(seq, outputfile) 
- 
-   ! release the sequence memory
-   call destroy_obs_sequence(seq)
+   num_inserted = combine_sequences(small_sequence, big_sequence, l1_files(ifile))
+
+   if (verbose > 2) then
+      call print_obs_seq_summary(small_sequence)
+   elseif (verbose > 1) then
+      write(string1,*)trim(l1_files(ifile))//' added ',num_inserted,' observations.'
+      call error_handler(E_MSG,string1,'')
+   endif
+
+   call destroy_obs_sequence(small_sequence)
 
 enddo
 
-call error_handler(E_MSG, source, 'Finished successfully.',source,revision,revdate)
+! Report on the current status of the entire observation sequence
+if (verbose > 0) then
+   call print_obs_seq_summary(big_sequence)
+endif
+
+! Write the sequence to a disk file
+if ( get_num_obs(big_sequence) > 0 )  then
+   call write_obs_seq(big_sequence, outputfile) 
+else
+   call error_handler(E_ERR,'NO OBSERVATIONS TO WRITE',source)
```

Comment:
i'm not sure i'd make this choice to error out if no obs are there.  in a big batch conversion, it's possible there are missing files, etc.  printing a msg is great but an error stops things abruptly.

Reply:
These filenames are pretty unpredictable so the only way I know of to create them is to capture a listing - so I'm not really concerned about missing files ... now ... files with no valid observations is another thing. I haven't run into that and am not sure what the intent is.  I'm willing to let the end user modify this to a message if they want. I don't want the default behavior to be to just blow by 'empty/bad' files - particularly in a batch job where no one actually looks at the output.

Reply:
in batch jobs i actually always grepped the output logs because real world obs files are such a crapshoot.  i don't agree with making this an error because if i've waited in the batch queue for an hour and i'm converting 30 days of data and the first one has a problem i get nothing for my trouble.  but like you said, the user can always modify the code.  i'd let this go if you add a comment above the error line saying something like "change E_ERR to E_MSG to allow this program to exit without returning an error code."

Reply:
I added comments and made the output string mention the comment in the code, so they know which one to edit.